pilot/cmd/pilot-agent/status/ready/probe.go

```diff
@@ -44,13 +44,31 @@ func (p *Probe) Check() error {
 
 	// Envoy has received some configuration, make sure that configuration has been received for
 	// all inbound ports.
-	if err := p.checkInboundConfigured(); err != nil {
+	if err := p.checkInbound(); err != nil {
 		return err
 	}
 
 	return p.checkServerInfo()
 }
 
+func (p *Probe) checkInbound() error {
+	if !p.ProbeByVirtualInboundListener || p.NodeType != model.SidecarProxy {
+		return p.checkInboundConfigured()
+	}
+	return p.checkInboundVirtualListener()
+
+}
+
+func (p *Probe) checkInboundVirtualListener() error {
+	// Warning: this is not reliable at envoy start up.
+	// TODO(silentdai): switch to /configdump
```

Comment:
why is config_dump more reliable?

Reply:
/listeners could return some listeners actually in warming... I want to create an issue to see if envoyproxy has the plan to fix the semantic of /listener

Reply:
Maybe `http://localhost:15000/listeners?format=json`? They improved it recently to have more info

Reply:
The drawback is referring to `/listener?format=json`. More info so that I can use listener name to located the 15006 listener. 
However, json and plain /listener suffer from the early-active.

That said, that bug/feature in envoyproxy affect us regardless with this PR or not

Reply:
sounds good. per above comment I don't think we need this anyways

Reply:
It's low priority since we survived but better leave the know issue here  :)